src/machine/board_rpi3.go

```diff
@@ -0,0 +1,182 @@
+	MiniUARTInterruptIdentify uint32 //0x48
+	MiniUARTLineControl       uint32 //0x4C
+	MiniUARTModemControl      uint32 //0x50
+	MiniUARTLineStatus        uint32 //0x54
+	MiniUARTModemStatus       uint32 //0x58
+	MiniUARTScratch           uint32 //0x5C
+	MiniUARTExtraControl      uint32 //0x60
+	MiniUARTExtraStatus       uint32 //0x64
+	MiniUARTBAUD              uint32 //0x68
+	reserved01                [5]uint32
+	SPI1ControlRegister0      uint32 //0x80
+	SPI1ControlRegister1      uint32 //0x84
+	SPI1Status                uint32 //0x88
+	reserved02                uint32 //0x8C
+	SPI1Data                  uint32 //0x90
+	SPI1Peek                  uint32 //0x94
+	reserved03                [10]uint32
+	SPI2ControlRegister0      uint32 //0xC0
+	SPI2ControlRegister1      uint32 //0xC4
+	SPI2Status                uint32 //0xC8
+	reserved04                uint32
+	SPI2Data                  uint32 //0xD0
+	SPI2Peek                  uint32 //0xD4
+}
+
+type GPIORegisterMap struct {
+	FuncSelect0              uint32 //0x00
+	FuncSelect1              uint32 //0x04
+	FuncSelect2              uint32 //0x08
+	FuncSelect3              uint32 //0x0C
+	FuncSelect4              uint32 //0x10
+	FuncSelect5              uint32 //0x14
+	reserved00               uint32 //0x18
+	OutputSet0               uint32 //0x1C
+	OutputSet1               uint32 //0x20
+	reserved01               uint32 //0x24
+	OutputClear0             uint32 //0x28
+	OutputClear1             uint32 //0x2C
+	reserved03               uint32 //0x30
+	Level0                   uint32 //0x34
+	Level1                   uint32 //0x38
+	reserved04               uint32 //0x3C
+	EventDetectStatus0       uint32 //0x40
+	EventDetectStatus1       uint32 //0x44
+	reserved05               uint32 //0x48
+	RisingEdgeDetectEnable0  uint32 //0x4C
+	RisingEdgeDetectEnable1  uint32 //0x50
+	reserved06               uint32 //0x54
+	FallingEdgeDetectEnable0 uint32 //0x58
+	FallingEdgeDetectEnable1 uint32 //0x5C
+	reserved07               uint32 //0x60
+	HighDetectEnable9        uint32 //0x64
+	HighDetectEnable1        uint32 //0x68
+	reserved08               uint32 //0x6C
+	LowDetectEnable0         uint32 //0x70
+	LowDetectEnable1         uint32 //0x74
+	reserved09               uint32 //0x78
+	AsyncRisingEdgeDetect0   uint32 //0x7C
+	AsyncRisingEdgeDetect1   uint32 //0x80
+	reserved0A               uint32 //0x84
+	AsyncFallingEdgeDetect0  uint32 //0x88
+	AsyncFallingEdgeDetect1  uint32 //0x8C
+	reserved0B               uint32 //0x90
+	PullUpDownEnable         uint32 // 0x94
+	PullUpDownEnableClock0   uint32 //0x98
+	PullUpDownEnableClock1   uint32 //0x9C
+	reserved0C               uint32 //0xA0
+	test                     uint32 //0xA4
+}
+
+type SysTimerRegisterMap struct {
+	ControlStatus uint32 //0x00
+	CounterLower32 uint32 //0x04
+	CounterHigher32 uint32 //0x08
+	reservedGPU0 uint32 //0x0C
+	Compare1 uint32 //0x10
+	reservedGPU2 uint32 //0x14
+	Compare3 uint32 //0x18
+}
+
+func UARTInit() {
```

Comment:
This UART should have an interface consistent with other targets. Something like:

```go
type UART struct {
    // You can leave this empty.
    // If there are multiple UARTs with a similar memory map (at different addresses)
    // you can include a pointer to the block of registers here.
}

func (uart UART) Configure(config UARTConfig) error {
    // this has the contents of UARTInit
}

func (uart UART) WriteByte(c byte) error {
    // this has the contents of UARTPutc
}
```

Reply:
Updated to reflect your UART defn.  However, right now it's pretty dumb because I'm using the (simplest) UART of the many on an RPI.  When I get to doing UART1 and so forth then it will be time to do it properly.